### PR TITLE
feat: add Franca imagenet benchmark

### LINKED ISSUE #2010: New SSL method: Franca (Nested Matryoshka Clustering)
## Overview

Franca (Venkataramanan et al., valeoai) looks like a good fit for lightly's DINO/DINOv2 family.

- **Paper:** https://arxiv.org/abs/2507.14137
- **Code:** https://github.com/valeoai/Franca (Apache-2.0)

## Franca contributions

It's a fully open DINOv2-based model (code, data, weights). It keeps the DINOv2 setup and adds:

- **Nested "Matryoshka" clustering**: the clustering head runs at several nested embedding sizes at once, giving coarse-to-fine assignments.
- **Cyclic-shift masking** for the iBOT branch.
- A post-hoc **positional-disentanglement head (RASA)**.

On ImageNet-21K, and separately a LAION subset, it reports matching or surpassing DINOv2, CLIP, and SigLIPv2.

## Fit with lightly

Most of it is the DINOv2 recipe already in lightly, so the main new pieces would be the Matryoshka head and loss, plus the cyclic-shift masking. I think we should leave out RASA as it's post-hoc (?)

Would it be of interest to include the method in lightly? If so I can try to implement it after agreeing on the implementation plan.

### PR DESCRIPTION
**Pull request by lorinczszabolcs**
Adds the ImageNet benchmark for Franca (#2010): `benchmarks/imagenet/vitb16/franca.py` and `franca` registered in `main.py`'s `METHODS`. Uses a ViT-S/16 backbone with explicit `pretrained=False`, the matryoshka DINO and iBOT heads, cyclic block masking, an `OnlineLinearClassifier`, AdamW with cosine warmup, and the EMA momentum schedule.

Franca releases only patch-14 models, so there is no ViT-S reference config. This trains ViT-S/16 from scratch in the shared vitb16 harness, following the lightly benchmark convention (like the CAPI benchmark's vit-small). LR, weight decay, and momentum follow the shared harness rather than Franca's exact values, for comparability with the other methods.

Script only. I can't run the ImageNet benchmark on my side, so no numbers are filled in yet. Verified that it imports and static checks pass.

Follow up of #2065. Closes #2010.

## Summary

- Added Franca with nested projection heads, DINO/iBOT losses, and cyclic block masking.
- Added ImageNet benchmark registration, documentation, and PyTorch/Lightning examples.
- Added unit tests for Franca heads, losses, and masking.
- No benchmark results are included.

### CODE REVIEW

File: docs/source/examples/models.rst [hunk 1]
```diff
@@ -20,6 +20,7 @@ for PyTorch and PyTorch Lightning to give you a headstart when implementing your
     dino.rst
     dinov2.rst
     fastsiam.rst
+    franca.rst
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply docs/source/examples/models.rst:23]: I ran `make -C docs html-noplot` on the new page and the toctree entry. The build succeeds and generates `examples/franca.html` with no new warnings.

File: examples/pytorch/franca.py [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+            nesting_dims=NESTING_DIMS,
+            output_dim=OUTPUT_DIM,
+        )
+
+        teacher_dino_head = franca_head()
+        student_dino_head = franca_head()
+
+        if ibot_separate_head:
+            teacher_ibot_head = franca_head()
+            student_ibot_head = franca_head()
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply examples/pytorch/franca.py:104]: This matches the existing lightly examples and benchmarks: `dinov2.py` and `capi.py` both deep-copy the backbone but construct the projection heads independently, so I kept Franca consistent with them.

File: examples/pytorch_lightning_distributed/franca.py [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+# Train with DDP and use Synchronized Batch Norm for a more accurate batch norm
+# calculation. Distributed sampling is also enabled with replace_sampler_ddp=True.
+trainer = pl.Trainer(
+    max_epochs=50,
+    devices="auto",
+    accelerator="gpu",
+    strategy="ddp_find_unused_parameters_true",
+    sync_batchnorm=True,
+    use_distributed_sampler=True,  # or replace_sampler_ddp=True for PyTorch Lightning <2.0
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply examples/pytorch_lightning_distributed/franca.py:299]: This follows the existing `dinov2.py` distributed example, which also uses `use_distributed_sampler=True` and documents the 1.x alternative in the inline comment (`replace_sampler_ddp=True for PyTorch Lightning <2.0`). I kept Franca consistent with it. Dropping 1.x support in the examples would be a repo-wide change rather than a Franca-specific one.

File: docs/source/examples/franca.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+.. tabs::
+    .. tab:: PyTorch
+
+        .. image:: /_static/colab-badge.svg
+            :target: https://colab.research.google.com/github/lightly-ai/lightly/blob/master/examples/notebooks/pytorch/franca.ipynb
+
+        This example can be run from the command line with::
+
+            python lightly/examples/pytorch/franca.py
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply docs/source/examples/franca.rst:40]: I kept the `lightly/` prefix to match every other method page (for example `dinov2.rst` documents `python lightly/examples/pytorch/dinov2.py`). Changing only this page would make it inconsistent with the rest of the docs. If the prefix should change, that is a repo-wide docs update rather than a Franca-specific one.

File: lightly/models/utils.py [hunk 5]
```diff
@@ -1055,6 +1055,99 @@ def random_block_mask_image(
...
+        if num_masked >= num_patches:
+            grid.fill_(True)
+        elif num_masked > 0:
+            # Sample the aspect ratio of the masked block, clamped so it fits.
+            min_lar = max(log_aspect_ratio[0], math.log(num_masked / width**2))
+            max_lar = min(
+                log_aspect_ratio[1], math.log(height**2 / (num_masked + 1e-5))
+            )
+            if min_lar > max_lar:
+                min_lar = max_lar
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply lightly/models/utils.py:1129]: This matches lightly's existing `random_inverse_block_mask`, which uses the same aspect clamping. The default range (min_aspect=0.5, max_aspect=2.0) always accommodates the requested count. The case here needs an extreme aspect range (0.05-0.1) that cannot geometrically fit num_masked in the grid, which is a degenerate configuration. The bounds are already validated (positive, min <= max), so I would rather keep it consistent with `random_inverse_block_mask` than add exact-count sizing the sibling does not have.

File: examples/notebooks/pytorch_lightning/franca.ipynb [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+    "        # Heads\n",
+    "        franca_head = partial(\n",
+    "            FrancaProjectionHead,\n",
+    "            input_dim=384,\n",
+    "            nesting_dims=NESTING_DIMS,\n",
+    "            output_dim=OUTPUT_DIM,\n",
+    "        )\n",
+    "\n",
+    "        teacher_dino_head = franca_head()\n",
+    "        student_dino_head = franca_head()\n",
```
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply examples/notebooks/pytorch_lightning/franca.ipynb:193]: Same as the Python source: `dinov2.py` and `capi.py` construct the projection heads independently, so I kept Franca consistent with them.

File: examples/pytorch/franca.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lorinczszabolcs) [reply examples/pytorch/franca.py:104]: @gabrielfruet, I would need your input on this one: I kept the teacher and student heads constructed independently to match `dinov2.py` and `capi.py`, but it is also possible to deep-copy the student head into the teacher across the examples and the benchmark if you prefer that for Franca.
